src/index.rs

```diff
@@ -1053,20 +1082,55 @@ impl Index {
             let sequence_number = sequence_number.value();
             sequence_number_to_inscription_entry
               .get(sequence_number)
-              .map(|entry| {
-                (
-                  sequence_number,
-                  InscriptionEntry::load(entry.unwrap().value()).id,
-                )
-              })
+              .map(|entry| InscriptionEntry::load(entry.unwrap().value()).id)
           })
           .map_err(|err| err.into())
       })
-      .collect::<Result<Vec<(u32, InscriptionId)>>>()?;
+      .skip(page_index.saturating_mul(page_size).try_into().unwrap())
+      .take(page_size.saturating_add(1).try_into().unwrap())
+      .collect::<Result<Vec<InscriptionId>>>()?;
+
+    let more = ids.len() > page_size.try_into().unwrap();
+
+    if more {
+      ids.pop();
+    }
+
+    Ok((ids, more))
+  }
+
+  pub(crate) fn get_inscription_id_by_sat_indexed(
+    &self,
+    sat: Sat,
+    inscription_index: isize,
+  ) -> Result<Option<InscriptionId>> {
+    let rtx = self.database.begin_read()?;
+
+    let sequence_number_to_inscription_entry =
+      rtx.open_table(SEQUENCE_NUMBER_TO_INSCRIPTION_ENTRY)?;
 
-    ids.sort_by_key(|(sequence_number, _id)| *sequence_number);
+    let sat_to_sequence_number = rtx.open_multimap_table(SAT_TO_SEQUENCE_NUMBER)?;
 
-    Ok(ids.into_iter().map(|(_sequence_number, id)| id).collect())
+    if inscription_index < 0 {
+      sat_to_sequence_number
+        .get(&sat.n())?
+        .nth_back(inscription_index.abs_diff(0))
```

Comment:
This should be tested, and I think you'll need to do `(inscription_index - 1).abs_diff(0)`

Reply:
should be +1